tools/integration_tests/read_cache/cache_file_for_include_regex_test.go

```diff
@@ -0,0 +1,165 @@
+	setup.SaveGCSFuseLogFileInCaseOfFailure(s.T())
+	setup.UnmountGCSFuseAndDeleteLogFile(rootDir)
+}
+
+////////////////////////////////////////////////////////////////////////
+// Test scenarios
+////////////////////////////////////////////////////////////////////////
+
+func (s *cacheFileForIncludeRegexTest) TestReadsForIncludedFile() {
+	if !s.cacheEnable {
+		return
+	}
+
+	testFileName := setupFileInTestDir(s.ctx, s.storageClient, fileSizeForRangeRead, s.T())
+
+	// Read the file and validate that it is cached.
+	expectedOutcome1 := readChunkAndValidateObjectContentsFromGCS(s.ctx, s.storageClient, testFileName, zeroOffset, s.T())
+	expectedOutcome2 := readChunkAndValidateObjectContentsFromGCS(s.ctx, s.storageClient, testFileName, offset1000, s.T())
+
+	structuredReadLogs := read_logs.GetStructuredLogsSortedByTimestamp(setup.LogFile(), s.T())
+	validate(expectedOutcome1, structuredReadLogs[0], true, false, 1, s.T())
+	validate(expectedOutcome2, structuredReadLogs[1], false, true, 1, s.T())
+	validateFileIsCached(testFileName, s.T())
+}
+
+func (s *cacheFileForIncludeRegexTest) TestReadsForNonIncludedFile() {
+	if s.cacheEnable {
+		return
+	}
+	testFileName := setupFileInTestDir(s.ctx, s.storageClient, fileSizeForRangeRead, s.T())
+
+	// Read the file and validate that it is not cached.
+	expectedOutcome1 := readChunkAndValidateObjectContentsFromGCS(s.ctx, s.storageClient, testFileName, zeroOffset, s.T())
+	expectedOutcome2 := readChunkAndValidateObjectContentsFromGCS(s.ctx, s.storageClient, testFileName, offset1000, s.T())
+
+	structuredReadLogs := read_logs.GetStructuredLogsSortedByTimestamp(setup.LogFile(), s.T())
+	validate(expectedOutcome1, structuredReadLogs[0], true, false, 1, s.T())
+	validate(expectedOutcome2, structuredReadLogs[1], false, false, 1, s.T())
+	validateFileIsNotCached(testFileName, s.T())
+}
+
+////////////////////////////////////////////////////////////////////////
+// Test Function (Runs once before all tests)
+////////////////////////////////////////////////////////////////////////
+
+func TestCacheFileForIncludeRegexTest(t *testing.T) {
+	ts := &cacheFileForIncludeRegexTest{ctx: context.Background()}
+	// Create storage client before running tests.
+	closeStorageClient := client.CreateStorageClientWithCancel(&ts.ctx, &ts.storageClient)
+	defer func() {
+		err := closeStorageClient()
+		if err != nil {
+			t.Errorf("closeStorageClient failed: %v", err)
+		}
+	}()
+
+	// Run tests for mounted directory if the flag is set.
+	if setup.AreBothMountedDirectoryAndTestBucketFlagsSet() {
+		ts.cacheEnable = true
+		suite.Run(t, ts)
+		return
+	}
+
+	// Run with cache directory pointing to RAM based dir
+	ramCacheDir := path.Join("/dev/shm", cacheDirName)
+
+	//cacheEnable=true - Test included file is cached
+	//cacheEnable=false - Test non-included file is not cached
+	tests := []struct {
+		name        string
+		flags       gcsfuseTestFlags
+		cacheEnable bool
+	}{
+		{
+			name: "Test included file is cached",
```

Comment:
We use testcases when we want to run the same test for multiple scenarios. in your case you have 2 tests and each one is run only for one testcase.

either combine those testcases and take the output as test case parameter or
do the test setup and do suite.run which should run both tests. no need of testcases.